packages/backend/apps/multitenancy/middleware.py

```diff
@@ -0,0 +1,63 @@
+from graphql_relay import from_global_id
+from django.utils.functional import SimpleLazyObject
+from .models import Tenant, TenantMembership
+
+
+def get_current_tenant(args):
+    """
+    Retrieves the current tenant based on the GraphQL arguments.
+
+    Args:
+    - args (dict): Dictionary containing GraphQL arguments.
+
+    Returns:
+    - Tenant or None: The current tenant or None if not found.
+    """
+    tenant_id = args.get("input").get('tenantId')
+    if not tenant_id:
+        # for the purpose of Tenant CRUD actions
+        tenant_id = args.get("input").get('id')
+    try:
+        id_type, pk = from_global_id(tenant_id)
+        if id_type == "TenantType":
+            return Tenant.objects.get(pk=pk)
+    except (Tenant.DoesNotExist, TypeError):
+        return None
+
+
+def get_current_user_role(info):
```

Comment:
I'm not a fan of passing `info` object here. It's name is very ambiguous. Could it be named `get_current_user_role_from_req(req: Request)` for example? `info.context` is a request object.

Reply:
I just refactored this middleware in next step so I won't change it in this PR if that is fine by you. I think that new approach is more clear.